content/dev/code/notifiers.md

```diff
@@ -3,420 +3,243 @@ title: Notifiers and Observers - About
 
-In procedural code (or outside of a class that `extends base`) use the global `$zco_notifier` object. For example, inside the `zen_mail` function this event is triggered, which allows a plugin to update the sent email format:
+Events are identified by a string-names and are triggered via call to a `notify` method by an _event issuer_.  You can see a [list of notifiers](/dev/code/notifiers_list) provided by the Zen Cart core for reference. 
 
-<pre>  $zco_notifier->notify('NOTIFY_EMAIL_DETERMINING_EMAIL_FORMAT', $to_email_address, $customers_email_format, $module);
-</pre>
+**Note**: Plugins (Zen Cart extensions) can also be _event issuers_.
 
-### Observe and Prosper
+The `notify` method takes the following inputs:
 
-To take advantage of notifiers, developers need to write some code to watch for them. Observers need to be written as a class. There's even a nice directory, `includes/classes/observers`, where developers can put these classes.
+|    Input Name     | Required? | Description                                                  |
+| :---------------: | :-------: | ------------------------------------------------------------ |
+|     $eventId      |    Yes    | The string 'name' of the event, e.g. `NOTIFIER_CART_ADD_CART_END`. |
+|      $param1      |    No     | A read-only variable, the format of which varies by the `$eventId`.  Defaults to an empty array. |
+| $param2 - $param9 |    No     | A collection of read-write variables, passed as a reference.  The notification issuer is, essentially, giving permission for an observer to update these variable.  Each variable's format (and presence) varies by the `$eventId` and defaults to `null`. |
 
-Lets take an example. Using the notifier mentioned above (`NOTIFIER_CART_ADD_CART_END`), how would I write a class that watched for that event?
+#### Class-Based Event Notifications
 
+Within a class that extends the Zen Cart `base` class, e.g.:
+
+```php
+class shopping_cart extends base
 ```
-<?php
- class myObserver extends base {
-   function __construct() {
-     $this->attach($this, array('NOTIFIER_CART_ADD_CART_END'));
-   }
-...
- }
+
+&hellip; events can be issued via the `$this` keyword:
+
+```php
+$this->notify('EVENT_NAME');
 ```
 
-As you can see we have defined a new class called myObserver and in the constructor function for that class (function myObserver) have attached this myObserver class to the event `NOTIFIER_CART_ADD_CART_END`.
+For example, the shopping cart class issues this event after an item has been added to the cart:
+```php
+$this->notify('NOTIFIER_CART_ADD_CART_END');
+```
 
-"Fine," I hear you saying, "but how do I actually do anything useful?"
+When a class issues a notification using the `$this` keyword, _all_ **public** class-variables are available for use by a watching observer.  If a class desires to issue a notification without that access, the global `$zco_notifier` can be used to issue that notification.
```

Comment:
I'm not sure I understand the 2nd sentence here.

Reply:
I agree, now reviewing what was written.  An observer could always use a `global` on the issuing class to retrieve those public variables.

Reply:
Agreed. Might also need to `global` on both ends if the calling source doesn't already have access to global scope.